src/url.js

```diff
@@ -92,15 +92,18 @@ export function getWinOrigin(win) {
  * testing by freezing the object.
  * @param {string} url
  * @param {boolean=} opt_nocache
+ *   Cache is always ignored on ESM builds, see https://go.amp.dev/pr/31594
  * @return {!Location}
  */
 export function parseUrlDeprecated(url, opt_nocache) {
   if (!a) {
     a = /** @type {!HTMLAnchorElement} */ (self.document.createElement('a'));
-    cache = self.__AMP_URL_CACHE || (self.__AMP_URL_CACHE = new LruCache(100));
+    cache = IS_ESM
```

Comment:
Can we do https://github.com/ampproject/amphtml/pull/31594/commits/5e4a64c410c23f5e54024519d41710c094f5097a#diff-9ae110c522ab0898b1c00a5aa2c38b02993d1731ad1cb49d31bb97c2a3f3dca2R99-R101 instead?

```js
export function parseUrlDeprecated(url, opt_nocache) {
  if (IS_ESM) {
    return /** @type {!Location} */ (new URL(url));
  }
  ...
}